pkg/reconciler/security/istio/eventpolicybinding/controller.go

```diff
@@ -0,0 +1,90 @@
+	pkgreconciler "knative.dev/pkg/reconciler"
+	"knative.dev/pkg/tracker"
+
+	securityapi "github.com/google/knative-gcp/pkg/apis/security"
+	securityv1alpha1 "github.com/google/knative-gcp/pkg/apis/security/v1alpha1"
+	eventpolicyinformer "github.com/google/knative-gcp/pkg/client/injection/informers/security/v1alpha1/eventpolicy"
+	eventpolicybindinginformer "github.com/google/knative-gcp/pkg/client/injection/informers/security/v1alpha1/eventpolicybinding"
+	httppolicyinformer "github.com/google/knative-gcp/pkg/client/injection/informers/security/v1alpha1/httppolicy"
+	httppolicybindinginformer "github.com/google/knative-gcp/pkg/client/injection/informers/security/v1alpha1/httppolicybinding"
+	bindingreconciler "github.com/google/knative-gcp/pkg/client/injection/reconciler/security/v1alpha1/eventpolicybinding"
+	"github.com/google/knative-gcp/pkg/reconciler"
+)
+
+const (
+	// ReconcilerName is the name of the reconciler
+	ReconcilerName      = "IstioEventPolicyBindings"
+	controllerAgentName = "istio-eventpolicybinding-controller"
+)
+
+var eventPolicyGVK = securityv1alpha1.SchemeGroupVersion.WithKind("EventPolicy")
+
+// NewController initializes the controller and is called by the generated code
+// Registers event handlers to enqueue events
+func NewController(
+	ctx context.Context,
+	cmw configmap.Watcher,
+) *controller.Impl {
+
+	eventPolicyBindingInformer := eventpolicybindinginformer.Get(ctx)
+	httpPolicyBindingInformer := httppolicybindinginformer.Get(ctx)
+	eventPolicyInformer := eventpolicyinformer.Get(ctx)
+	httpPolicyInformer := httppolicyinformer.Get(ctx)
+
+	r := &Reconciler{
+		Base:                    reconciler.NewBase(ctx, controllerAgentName, cmw),
+		eventPolicyLister:       eventPolicyInformer.Lister(),
+		httpPolicyLister:        httpPolicyInformer.Lister(),
+		httpPolicyBindingLister: httpPolicyBindingInformer.Lister(),
+	}
+
+	impl := bindingreconciler.NewImpl(ctx, r)
+
+	r.Logger.Info("Setting up event handlers")
+
+	eventPolicyBindingInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: pkgreconciler.AnnotationFilterFunc(securityapi.PolicyBindingClassAnnotationKey, securityapi.IstioPolicyBindingClassValue, false),
+		Handler:    controller.HandleAll(impl.Enqueue),
+	})
+
+	httpPolicyBindingInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: controller.FilterGroupKind(securityv1alpha1.Kind("EventPolicyBinding")),
+		Handler:    controller.HandleAll(impl.EnqueueControllerOf),
+	})
+
+	httpPolicyInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: controller.FilterGroupKind(securityv1alpha1.Kind("EventPolicyBinding")),
+		Handler:    controller.HandleAll(impl.EnqueueControllerOf),
+	})
+
+	r.policyTracker = tracker.New(impl.EnqueueKey, controller.GetTrackerLease(ctx))
+	eventPolicyInformer.Informer().AddEventHandler(controller.HandleAll(
+		controller.EnsureTypeMeta(r.policyTracker.OnChanged, eventPolicyGVK),
```

Comment:
nit: can the eventPolicyGVK be inlined here for easier readability, or you are using it from somewhere else?

Reply:
NVM saw you are using it later